Mathlib/Analysis/SpecialFunctions/Log/ENNRealLogExp.lean

```diff
@@ -0,0 +1,170 @@
+/-
+Copyright (c) 2024 Rémy Degenne. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damien Thomine, Pietro Monticone, Rémy Degenne, Lorenzo Luccioli
+-/
+import Mathlib.Analysis.SpecialFunctions.Log.ENNRealExp
+import Mathlib.Analysis.SpecialFunctions.Log.ENNRealLog
+import Mathlib.MeasureTheory.Constructions.BorelSpace.Basic
+
+/-!
+# Properties of the extended logarithm and exponential
+
+We prove that `log` and `exp` define an order isomorphism between `ℝ≥0∞` and `EReal`.
+
+## Main Definitions
+- `ENNReal.logOrderIso`: The order isomorphism between `ℝ≥0∞` and `EReal` defined by `log`
+and `exp`.
+- `ENNReal.log_homeomorph`: `log` as a homeomorphism.
+- `ENNReal.exp_homeomorph`: `exp` as a homeomorphism.
```

Comment:
```suggestion
- `EReal.exp_homeomorph`: `exp` as a homeomorphism.
```
Also, it feels weird to have the order_iso version of the log but not of the exponential.

Reply:
`logOrderIso` actually contains the order isomorphism version of the exponential as the inverse function:
https://github.com/leanprover-community/mathlib4/blob/7ba4a1a7efa52aa3873d0ba28eb391c73280e7ea/Mathlib/Analysis/SpecialFunctions/Log/ENNRealLogExp.lean#L115-L120

Should we rename it to something like `logExpOrderIso` so it is more evident? Or do you mean that we should actually define an `expOrderIso` as a separate structure (or maybe as `logOrderIso.symm`)?